### DISCUSSION TITLE
Allow a role to create roles and assign permissions

### DISCUSSION THREAD

**Discussion by abdulrehman-dar** [spatie/laravel-permission#1924] [Q&A]
I have a system(CRM realestates) where we can have multiple roles. For example Admin, Agency, Agent. And I want to give access agencies to create there own roles and assign permissions(added by admin cannot create own). Is it is possible using spatie/permissions. If it is how can we do this need advice. The problem is how can we differ which role is for which agency and which agency assign user which permissions.

**Comment by erikn69** [comment]
Read documentation https://github.com/spatie/laravel-permission/blob/main/docs/basic-usage/teams-permissions.md
Or read other discussions with the solution instead of open a new one
